pipeline/load.py

```diff
@@ -0,0 +1,100 @@
+# pylint: disable=E1101
+"""Script that will insert plant data into RDS."""
+import os
+from typing import List, Tuple
+import pymssql
+import pandas as pd
+from dotenv import load_dotenv
+import transform as tf
+
+
+def get_connection() -> object:
+    """Gets connection to Microsoft SQL server"""
+    return pymssql.connect(database=os.getenv("DB_NAME"),
+                           user=os.getenv("DB_USER"),
+                           server=os.getenv("DB_HOST"),
+                           password=os.getenv("DB_PASSWORD"))
+
+
+def get_cursor(connection: object) -> object:
+    """Gets cursor"""
+    return connection.cursor()
+
+
+def get_foreign_key(db_cursor: object, table_name: str,
+                    column_name: str, value: str) -> int:
+    """Gets foreign keys."""
+    db_cursor.execute(
+        f"SELECT * FROM {os.getenv('SCHEMA_NAME')}.{table_name} WHERE {column_name} = %s", (value,))
+    result = db_cursor.fetchone()
+    if result:
+        return result[0]
+    print(f"Querying {table_name} for {column_name} = {value}")
+    raise ValueError('Invalid Data!')
+
+
+def get_all_plant_foreign_keys(db_cursor: object,
+                               dataframe: pd.DataFrame) -> List[Tuple[int, int, int]]:
+    """Gets all foreign keys for the plant table,"""
+    foreign_keys = []
+    for index, row in dataframe.iterrows():
+        try:
+            botanist_id = get_foreign_key(
+                db_cursor, 'botanist', 'botanist_email', row['botanist_email'])
+            species_id = get_foreign_key(
+                db_cursor, 'species', 'plant_name', row['plant_name'])
+            country_id = get_foreign_key(
+                db_cursor, 'country', 'country_code', row['country_code'])
+            foreign_keys.append((botanist_id, species_id, country_id))
```

Comment:
just a thought - if we wanted to speed this up in future, could using pandas .apply instead of .iterrows be a potential change?

Reply:
not suggesting this change now btw - just food for thought if we potentially want to optimise later on